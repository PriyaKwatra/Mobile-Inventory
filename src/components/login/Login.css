
.form {
    margin: auto;
    width: 50%;
    padding: 10px;
    position: relative;
    top:100px
  }

