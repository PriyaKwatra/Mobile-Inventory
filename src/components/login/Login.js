import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import { withRouter } from "react-router-dom";
import { setLoggedIn } from '../../actions/index';
import './Login.css';


export class Login extends Component {


    state = {
        userName: " ",
        password: " "
    };


    handleUserName = (event) => {
        this.setState({ userName: event.target.value });
    }

    handlePassword = (event) => {
        this.setState({ password: event.target.value });
    }

    handleSubmit = (event) => {

        if (this.state.userName === this.props.userName && this.state.password === this.props.password) {
            this.login();
        }

        else {
            alert("incorrect userName or password");
        }

    }

    login() {
        this.props.setLoggedIn();
        this.props.history.push("/");
    }

    render() {
        return (
            <div className="form">
                <h1 style={{ position: 'absolute', left: '250px' }} >Sign in</h1>
                <div style={{ position: 'absolute', top: '100px', left: '100px', width: '50%', padding: '20px' }}>
                    <p>UserName
          <input type="userName" className={`form-control mb-4 user`} placeholder="UserName"
                            name="userName" required onChange={this.handleUserName}  />
                    </p>
                    <p>Password
          <input type="password" className={`form-control mb-4 password`} placeholder="Password"
                            name="password" required onChange={this.handlePassword}  />
                    </p>
                    <div class="d-flex justify-content-around">
                    </div>
                    <Button variant="primary" color="info" block="true" onClick={this.handleSubmit} className="sign" >Sign in</Button>
                </div>
            </div>


        )
    }

}

const mapStateToProps = (state) => {
    return {
        userName: state.loggingInformation.userName,
        password: state.loggingInformation.password
    }
}

const mapDispatchToProps = (dispatch) => ({
    setLoggedIn: () => dispatch(setLoggedIn())
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login))