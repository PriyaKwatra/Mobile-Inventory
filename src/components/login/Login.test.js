import { shallow } from 'enzyme';
import React from 'react';
import { Login } from './Login';
import { mount } from 'enzyme';



describe('Login', () => {
    it('should render correctly in "debug" mode', () => {
        mount(<Login />)
    });
});

describe('Login', () => {
    it('Should not login when incorrect username and correct password', () => {
        const setLoggedIn = jest.spyOn(Login.prototype, 'login');
        const component = mount(<Login userName="a" password="b" />)
        component.find('.user').first().simulate('change', { target: { value: 'incorrect' } });
        component.find('.password').first().simulate('change', { target: { value: 'b' } });;
        component.find('.sign').first().simulate("click");
    });
});

describe('Login', () => {
    it('Should not login when incorrect username and incorrect password', () => {
        const setLoggedIn = jest.spyOn(Login.prototype, 'login');
        const component = mount(<Login userName="a" password="b" />)
        component.find('.user').first().simulate('change', { target: { value: 'incorrect' } });
        component.find('.password').first().simulate('change', { target: { value: 'incorrect' } });;
        component.find('.sign').first().simulate("click");
    });
});

describe('Login', () => {
    it('Should login when correct username and correct password', () => {
        const setLoggedIn = jest.spyOn(Login.prototype, 'login');
        const component = mount(<Login userName="a" password="b" setLoggedIn={() => { }} history={[]} />)
        component.find('.user').first().simulate('change', { target: { value: 'a' } });
        component.find('.password').first().simulate('change', { target: { value: 'b' } });;
        component.find('.sign').first().simulate("click");
        expect(setLoggedIn).toHaveBeenCalled();
    });
});



