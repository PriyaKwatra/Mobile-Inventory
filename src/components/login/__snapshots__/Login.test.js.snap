// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly in "debug" mode 1`] = `
<div
  className="form"
>
  <h1
    style={
      Object {
        "left": "250px",
        "position": "absolute",
      }
    }
  >
    Sign in
  </h1>
  <div
    style={
      Object {
        "left": "100px",
        "padding": "20px",
        "position": "absolute",
        "top": "100px",
        "width": "50%",
      }
    }
  >
    <p>
      UserName
      <input
        class="form-control mb-4"
        name="userName"
        onChange={[Function]}
        placeholder="UserName"
        required={true}
        type="userName"
      />
    </p>
    <p>
      Password
      <input
        class="form-control mb-4"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </p>
    <div
      class="d-flex justify-content-around"
    />
    <Button
      active={false}
      block="true"
      color="info"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Sign in
    </Button>
  </div>
</div>
`;

exports[`Login should render correctly in "debug" mode 1`] = `
<div
  className="form"
>
  <h1
    style={
      Object {
        "left": "250px",
        "position": "absolute",
      }
    }
  >
    Sign in
  </h1>
  <div
    style={
      Object {
        "left": "100px",
        "padding": "20px",
        "position": "absolute",
        "top": "100px",
        "width": "50%",
      }
    }
  >
    <p>
      UserName
      <input
        class="form-control mb-4"
        name="userName"
        onChange={[Function]}
        placeholder="UserName"
        required={true}
        type="userName"
      />
    </p>
    <p>
      Password
      <input
        class="form-control mb-4"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </p>
    <div
      class="d-flex justify-content-around"
    />
    <Button
      active={false}
      block="true"
      color="info"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Sign in
    </Button>
  </div>
</div>
`;
